### PR TITLE
Create workflow for publishing to private PyPI

### PR DESCRIPTION
**Pull request by mheath-oqc**

### CODE REVIEW

**Comment by mheath-oqc** [comment]
I have read the Contributor License Agreement and I hereby accept the Terms.

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
+name: Publish to OQC Private PyPI
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - '[0-9]+.[0-9]+.[0-9]+'
+  workflow_dispatch:
```
Comment (bgsach) [inline .github/workflows/publish.yml:7]: I believe this triggers on any push to main OR any tag push not AND! (we have this problem on qat at the moment).

**Review by keriksson-rosenqvist-oqc — CHANGES_REQUESTED** [review]
I would much rather see this centralised in the https://github.com/oqc-tech/oqc-actions repo

**Comment by mheath-oqc** [comment]
> I would much rather see this centralised in the https://github.com/oqc-tech/oqc-actions repo

I do agree and will be happy to help get this done as part of future work.

**Review by twinchester-oqc — APPROVED** [review]

**Review by bgsach — APPROVED** [review]

**Pull request merged**